docs/contributing/tests.md

```diff
@@ -0,0 +1,15 @@
+# Tests
+
+Tests are critical to Cursorless. Our tests fall broadly into a three categories:
+
+- **Recorded tests** are end-to-end / subcutaneous tests that run a full Cursorless command and check that the ide ends up in the correct state. These represent the bulk of our tests, and are generated automatically using the [test case recorder](./test-case-recorder.md)
```

Comment:
```suggestion
- **Recorded tests** are end-to-end / subcutaneous tests that run a full Cursorless command and check that the IDE ends up in the correct state. These represent the bulk of our tests, and are generated automatically using the [test case recorder](./test-case-recorder.md)
```

Reply:
lol, I love the use of "subcutaneous". I had not heard that before. Is the meaning from this? https://martinfowler.com/bliki/SubcutaneousTest.html

Actually, after reading the article, I'm a bit unsure if these are subcutaneous, or fully end to end. The article implies subcutaneous tests skip some part of the end-to-end UI testing. But the recorded tests do run in the full VS Code UI. I guess you could argue that they skip the speaking and processing of the Talon command? Otherwise I don't actually think we are skipping very much of the process (imo, a good thing).

Reply:
Yeah that's why I say end-to-end / subcutaneous here because they're kind of borderline. I could argue either direction, as you've done here 😄